pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFormatPatternSpec.java

```diff
@@ -81,6 +82,23 @@ public DateTimeFormatter getDateTimeFormatter() {
     return _dateTimeFormatter;
   }
 
+  /**
+   * Validates the sdf pattern
+   */
+  public static void validateFormat(@Nonnull String sdfPatternWithTz) {
+    String sdfPattern;
+    DateTimeZone dateTimeZone = DEFAULT_DATETIMEZONE;
+    Matcher m = SDF_PATTERN_WITH_TIMEZONE.matcher(sdfPatternWithTz);
+    if (m.find()) {
+      sdfPattern = m.group(SDF_PATTERN_GROUP).trim();
+      String timezoneString = m.group(TIMEZONE_GROUP).trim();
+      dateTimeZone = DateTimeZone.forTimeZone(TimeZone.getTimeZone(timezoneString));
+    } else {
+      sdfPattern = sdfPatternWithTz;
+    }
+    DateTimeFormat.forPattern(sdfPattern).withZone(dateTimeZone);
```

Comment:
Suggest wrapping it in a try-catch, and put some useful exception message such as `"Invalid SDF pattern: " + sdfPatternWithTz` for user to better understand the issue